source/incqueryserver-jupyter/iqs_jupyter/core_extensions.py

```diff
@@ -386,8 +390,29 @@ def __init__(
         else:
             fields = [self.address_field]
         
-        self.box=widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
-        if auto_display: self.display()
+        self.box = widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
+        if auto_display:
+            self.display()
+
+        if login_button:
+            btn_login = interact.options(manual=True, manual_name="Login")
+
+            @btn_login
```

Comment:
Also, is there a specific reason you use `interact` instead of just creating a Button widget and adding it to the box? Did you find it more concise this way?

Also, won't `interact` auto-display the button even if  the user sets `auto_display` in the `IQSConnectorWidget` constructor to `False`?

Reply:
Yes, using `interact` will display the button even if `auto_display` is set to `False`. I may use a Button widget instead.